subsys/net/ip/net_pkt.c

```diff
@@ -1180,40 +1214,27 @@ u16_t net_pkt_append(struct net_pkt *pkt, u16_t len, const u8_t *data,
 	}
 
 	/* Make sure we don't send more data in one packet than
-	 * MTU allows.
-	 * This check really belongs to net_pkt_append_bytes(),
-	 * but instead done here for efficiency, we assume
-	 * (at least for now) that that callers of
-	 * net_pkt_append_bytes() are smart enough to not
-	 * overflow MTU.
+	 * protocol or MTU allows.
 	 */
-	max_len = 0x10000;
+	max_len = pkt->data_len;
```

Comment:
it could anyway be called here everytime and available space would be computed from that minus current total length.

no?

Reply:
We need to remember the amount of space remaining so that two consecutive calls to net_pkt_append() succeed, unless there is another application data counter somewhere in the struct already?

Reply:
As @pfl mentioned, we need to support the case where multiple append/insert calls are done, so the amount of bytes needs to be remembered.

Reply:
But we know that: we can get the actual occupied length in the chain of net_buf.

Reply:
What variables would we need to look at?

Reply:
I though the whole length of fragment list, net_buf_frags_len(pkt->frags), could help. But as the ll reserve part is badly managed for now, it won't work.